sync-global-pullsecret/sync-global-pullsecret.go

```diff
@@ -0,0 +1,257 @@
+var writeFileFunc = os.WriteFile
+
+// GlobalPullSecretReconciler reconciles a Secret object
+type GlobalPullSecretReconciler struct {
+	client.Client
+	Scheme                *runtime.Scheme
+	kubeletConfigJsonPath string
+	globalPSSecretName    string
+	globalPSSecretNS      string
+}
+
+// NewRunCommand creates a new cobra.Command for the sync-global-pullsecret command
+func NewRunCommand() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "sync-global-pullsecret",
+		Short: "Syncs a mixture between the user provided pull secret in DataPlane and the HostedCluster PullSecret to be deployed in the nodes of the HostedCluster",
+		Long:  `Syncs a mixture between the user provided pull secret in DataPlane and the HostedCluster PullSecret to be deployed in the nodes of the HostedCluster. The resulting pull secret is deployed in a DaemonSet in the DataPlane that updates the kubelet.config.json file with the new pull secret. If there are conflicting entries in the resulting global pull secret, the user provided pull secret will prevail.`,
+	}
+
+	opts := syncGlobalPullSecretOptions{
+		kubeletConfigJsonPath: defaultKubeletConfigJsonPath,
+	}
+	cmd.Flags().StringVar(&opts.globalPSSecretName, "global-pull-secret-name", defaultGlobalPSSecretName, "The name of the global pullSecret secret in the DataPlane.")
+	cmd.Run = func(cmd *cobra.Command, args []string) {
+		setupLog := ctrl.Log.WithName("global-pullsecret")
+		zapOpts := zap.Options{
+			Development: true,
+		}
+		ctrl.SetLogger(zap.New(zap.UseFlagOptions(&zapOpts)))
+		ctx := ctrl.SetupSignalHandler()
+		if err := opts.run(ctx); err != nil {
+			setupLog.Error(err, "unable to start manager")
+			os.Exit(1)
+		}
+	}
+
+	return cmd
+}
+
+// run executes the main logic of the sync-global-pullsecret command
+func (o *syncGlobalPullSecretOptions) run(ctx context.Context) error {
+	// Create manager
+	mgr, err := ctrl.NewManager(ctrl.GetConfigOrDie(), ctrl.Options{
+		Scheme: hyperapi.Scheme,
+		Cache: cache.Options{
+			DefaultNamespaces: map[string]cache.Config{defaultGlobalPullSecretNamespace: {}},
+		},
+	})
+	if err != nil {
+		return fmt.Errorf("failed to create manager: %w", err)
+	}
+
+	// Create reconciler
+	r := &GlobalPullSecretReconciler{
+		Client:                mgr.GetClient(),
+		Scheme:                mgr.GetScheme(),
+		kubeletConfigJsonPath: o.kubeletConfigJsonPath,
+		globalPSSecretName:    o.globalPSSecretName,
+		globalPSSecretNS:      defaultGlobalPullSecretNamespace,
+	}
+
+	// Create controller
+	if err := ctrl.NewControllerManagedBy(mgr).
+		For(&corev1.Secret{}).
+		WithEventFilter(predicate.Funcs{
+			// Adding filters to avoid processing events that are not relevant
+			CreateFunc: func(e event.CreateEvent) bool {
+				return o.isTargetSecret(e.Object)
+			},
+			UpdateFunc: func(e event.UpdateEvent) bool {
+				return o.isTargetSecret(e.ObjectNew)
+			},
+			DeleteFunc: func(e event.DeleteEvent) bool {
+				return false
```

Comment:
why are deleted events dropped?
what happens during deletion?

Reply:
HCCO will recreate it with the right content